### PR TITLE
Fix for deleted Reduce or Zero rate tax classes

### PR DESCRIPTION
**Pull request by geektzu**
### All Submissions:

* [X] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR is to fix an existing issue with default tax classes being re-created after a version upgrade:
https://github.com/woocommerce/woocommerce/issues/29708

### How to test the changes in this Pull Request:

1.  Remove WooCommerce plugin completely (deactivate + delete)
2. Reinstall v4.5 by uploading the zip file downloaded from https://developer.woocommerce.com/releases/
3. Unzip 5.3.0 zip, modify the lines as per solution above and re-create zip file
4. Upload for testing
5. Screen capture video: https://drive.google.com/file/d/1L9G4aXmtDo4PM8ueUJbeDr-4nCHuOHIB/view?usp=sharing

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Dev - Added an if condition block to check for new install before creating Zero and Reduced rate tax classes in class-wc-install.php

### CODE REVIEW

**Comment by geektzu** [comment]
Passed unit test locally. 

geektzu@geektzu-iMac:~/Documents/GitHub/woocommerce$ vendor/bin/phpunit --coverage-text
Installing...
Running as single site... To run multisite, use -c tests/phpunit/multisite.xml
Installing WooCommerce...
Not running ajax tests. To execute these, use --group ajax.
Not running ms-files tests. To execute these, use --group ms-files.
Not running external-http tests. To execute these, use --group external-http.
PHPUnit 6.5.14 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.3
Configuration: /home/geektzu/Documents/GitHub/woocommerce/phpunit.xml
Error:         No code coverage driver is available

.............................................................   61 / 1819 (  3%)
.............................................................  122 / 1819 (  6%)
.............................................................  183 / 1819 ( 10%)
.............................................................  244 / 1819 ( 13%)
.............................................................  305 / 1819 ( 16%)
.............................................................  366 / 1819 ( 20%)
.............................................................  427 / 1819 ( 23%)
.............................................................  488 / 1819 ( 26%)
.............................................................  549 / 1819 ( 30%)
.............................................................  610 / 1819 ( 33%)
.............................................................  671 / 1819 ( 36%)
.............................................................  732 / 1819 ( 40%)
.............................................................  793 / 1819 ( 43%)
.............................................................  854 / 1819 ( 46%)
.............................................................  915 / 1819 ( 50%)
.............................................................  976 / 1819 ( 53%)
............................................................. 1037 / 1819 ( 57%)
............................................................. 1098 / 1819 ( 60%)
............................................................. 1159 / 1819 ( 63%)
............................................................. 1220 / 1819 ( 67%)
............................................................. 1281 / 1819 ( 70%)
............................................................. 1342 / 1819 ( 73%)
............................................................. 1403 / 1819 ( 77%)
............................................................. 1464 / 1819 ( 80%)
............................................................. 1525 / 1819 ( 83%)
............................................................. 1586 / 1819 ( 87%)
............................................................. 1647 / 1819 ( 90%)
............................................................. 1708 / 1819 ( 93%)
............................................................. 1769 / 1819 ( 97%)
..................................................            1819 / 1819 (100%)

You should really fix these slow tests (>500ms)...
 1. 48160ms to run WC_Tests_API_Product:test_products_filter_post_status
 2. 37031ms to run WC_Stock_Functions_Tests:test_status_transition_stock_no_effect_to_stock_no_effect
 3. 21161ms to run WC_Tests_REST_System_Status_V2:test_get_system_status_tool_without_permission
 4. 20637ms to run WC_Tests_Cart:test_get_cart_item_quantities
 5. 16749ms to run WC_Tests_API_Orders_V2:test_update_order_remove_items
 6. 8733ms to run Products_API_V2:test_products_batch
 7. 8492ms to run WC_Stock_Functions_Tests:test_wc_get_low_stock_amount_simple_set
 8. 5973ms to run WC_Tests_Paypal_Gateway_Request:test_request_url
 9. 4483ms to run WC_Tests_API_Orders:test_update_order_invalid_id
 10. 3644ms to run WC_Tests_REST_System_Status_V2:test_get_system_status_info_returns_root_properties
...and there are 241 more above your threshold hidden from view

Time: 11.17 minutes, Memory: 133.00MB

OK (1819 tests, 6963 assertions)

Setting up E2E testing environment locally now.

**Comment by geektzu** [comment]
@ObliviousHarmony, Hi Chris, would appreciate your help to review the code. It is my first time contributing to woocommerce. Thank you in advance.

File: includes/class-wc-install.php [hunk 1]
```diff
@@ -620,8 +620,11 @@ private static function create_options() {
 		add_option( 'woocommerce_demo_store', 'no', '', 'no' );
 
 		// Define initial tax classes.
-		WC_Tax::create_tax_class( __( 'Reduced rate', 'woocommerce' ) );
-		WC_Tax::create_tax_class( __( 'Zero rate', 'woocommerce' ) );
+		if ( self::is_new_install() ) {
+			// Moved behind only for new install.
```
Comment (roykho) [inline includes/class-wc-install.php:624]: Can you please keep the original comment here on this line. Your added comment isn't needed as the conditional is clear enough to state what is going on.

**Comment by geektzu** [comment]
Updated the comments as per code review.
Re-ran Code Snipper and Unit Test. All checks passed as below:

geektzu@geektzu-iMac:~/Documents/GitHub/woocommerce$ ./vendor/bin/phpcs --encoding=utf-8 -s -n -p --report-full --report-checkstyle=./phpcs-report.xml ./includes/class-wc-install.php 
**. 1 / 1 (100%)**


Time: 665ms; Memory: 12MB

geektzu@geektzu-iMac:~/Documents/GitHub/woocommerce$ vendor/bin/phpunit 
Installing...
Running as single site... To run multisite, use -c tests/phpunit/multisite.xml
Installing WooCommerce...
Not running ajax tests. To execute these, use --group ajax.
Not running ms-files tests. To execute these, use --group ms-files.
Not running external-http tests. To execute these, use --group external-http.
PHPUnit 6.5.14 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.3
Configuration: /home/geektzu/Documents/GitHub/woocommerce/phpunit.xml

.............................................................   61 / 1819 (  3%)
.............................................................  122 / 1819 (  6%)
.............................................................  183 / 1819 ( 10%)
.............................................................  244 / 1819 ( 13%)
.............................................................  305 / 1819 ( 16%)
.............................................................  366 / 1819 ( 20%)
.............................................................  427 / 1819 ( 23%)
.............................................................  488 / 1819 ( 26%)
.............................................................  549 / 1819 ( 30%)
.............................................................  610 / 1819 ( 33%)
.............................................................  671 / 1819 ( 36%)
.............................................................  732 / 1819 ( 40%)
.............................................................  793 / 1819 ( 43%)
.............................................................  854 / 1819 ( 46%)
.............................................................  915 / 1819 ( 50%)
.............................................................  976 / 1819 ( 53%)
............................................................. 1037 / 1819 ( 57%)
............................................................. 1098 / 1819 ( 60%)
............................................................. 1159 / 1819 ( 63%)
............................................................. 1220 / 1819 ( 67%)
............................................................. 1281 / 1819 ( 70%)
............................................................. 1342 / 1819 ( 73%)
............................................................. 1403 / 1819 ( 77%)
............................................................. 1464 / 1819 ( 80%)
............................................................. 1525 / 1819 ( 83%)
............................................................. 1586 / 1819 ( 87%)
............................................................. 1647 / 1819 ( 90%)
............................................................. 1708 / 1819 ( 93%)
............................................................. 1769 / 1819 ( 97%)
................................................^[[B..            1819 / 1819 (100%)

You should really fix these slow tests (>500ms)...
 1. 23849ms to run WC_Tests_API_Reports_Coupons_Totals:test_register_routes
 2. 20500ms to run WC_Tests_API_Coupons:test_get_coupon_without_permission
 3. 16399ms to run WC_Tests_API_Shipping_Zones:test_get_methods_invalid_zone_id
 4. 16113ms to run WC_Tests_API_Orders_V2:test_delete_order_without_permission
 5. 14437ms to run WC_Tests_REST_System_Status_V2:test_get_system_status_info_returns_root_properties
 6. 12203ms to run WC_Tests_Discounts:test_is_coupon_valid_fixed_cart_sale_items
 7. 8972ms to run Shipping_Methods_V2:test_register_routes
 8. 8588ms to run Customers_V2:test_register_routes
 9. 7881ms to run WC_REST_Products_Controller_Tests:test_products_get_each_field_one_by_one
 10. 5444ms to run WC_Tests_Paypal_Gateway_Request:test_request_url
...and there are 251 more above your threshold hidden from view

Time: 10.41 minutes, Memory: 133.00MB

**OK (1819 tests, 6963 assertions)**

File: includes/class-wc-install.php (hunk 1 above)
Comment (geektzu) [reply includes/class-wc-install.php]: @roykho please review again

**Review by roykho — APPROVED** [review]
LGTM! Thanks for the PR!

**Pull request merged**